_posts/2020-06-23-sourcify-faq.md

```diff
@@ -0,0 +1,195 @@
+published files. By doing so, it can be verified that the contract code is exactly the same as when it was deployed on 
+blockchain and has not been adjusted or tampered with before deploying it.
+
+### Why is source verification important?
+
+Without open source contracts it's hard to tell what the application you are using will actually be doing. 
+One of the decentralized web's biggest aspriations is to build a trustless infrastructure, aiming to reduce the need to 
+trust third parties and service providers and decreasing censorship opportunities. It should, by design, not be required to 
+"blindly" trust a dapp developer or web3 project. It shall, in the best case, always be possible to verify it for yourself, 
+or if you're not tech savy enough, leverage the power of the crowd of other developers which can look at and call 
+out malicious code once it's open source.
+
+### Why should I publish my source code / open-source it?
+
+Closed source contract code cannot easily be verified since the trustless setup is not maintained if you have to rely 
+on a third party which verifies the code for you. In that scenario, the verification service providers could be an attack 
+vector and would need to be trusted in.
+
+## Metadata 
+
+### What is bytecode metadata?
+
+Metadata is data that describes and gives information about other data. [Bytecode (contract) metadata](https://solidity.readthedocs.io/en/latest/metadata.html) 
+in Solidity is supported since version 0.4.7 and is automatically generated by the Solidity compiler in form of a JSON file. 
+The metadata file contains information about the compiled contract, e.g. the used compiler version, the sources used, 
+the ABI and the NatSpec documentation. The metadata file also contains IPFS and SWARM hashes of the source code, which allows 
+to retrieve them if they are available.
+
+Metadata files can also contain literal sources, which means that the acutal code of the contract is present in the metadata 
+file's source section. Literal sources will be covered by the Sourcify verification [in future](https://github.com/ethereum/sourcify/issues/187), too.
+
+### What is NatSpec?
+
+[NatSpec](https://solidity.readthedocs.io/en/latest/natspec-format.html) is short for Ethereum Natural Language Specification 
+Format. Solidity contracts can use NatSpec comments to provide rich documentation for functions, return variables and more. 
+The comments are categorized by developer-focused and end-user-focused messages. The end-user-focused messages can be crucial 
+to users since they describe what the contract interaction will do in easy language. It is highly recommended to fully 
+annotate Solidity contracts using NatSpec for all public interfaces. 
+
+NatSpec comments can be displayed to the user at the time of interaction, e.g. by being promted in the wallet interface 
+before signing a transaction.
```

Comment:
Or as a comment in a list of past transactions

Reply:
Good point, I'll add it!